task-impls/src/consensus.rs

```diff
@@ -461,14 +628,114 @@ where
                                 );
                             }
                         }
-                        // self.update_view(view);
+
+                        self.high_qc = leaf.justify_qc.clone();
+
+                        //             let mut new_anchor_view = consensus.last_decided_view;
```

Comment:
Yep this was the logic I knew was missing I'll work off this to get it to compile/mostly work before the pair programming hopefully